src/js/view/components/timeslider.js

```diff
@@ -197,7 +209,18 @@ define([
             this.showThumbnail(time);
 
             utils.addClass(this.timeTip.el, 'jw-open');
-            this.timeTip.el.style.left = (pct*100) + '%';
+
+            var timeTipWidth = this.timeTip.width();
+            var widthPct = _railBounds.width / 100;
+            var tolerance = playerWidth - _railBounds.width;
+            var timeTipPct = 0;
+            if (timeTipWidth > tolerance) {
+                // timeTip may go outside the bounds of the player. determine the % of tolerance needed
+                timeTipPct = (timeTipWidth - tolerance) / (2 * 100 * widthPct);
+            }
+            var safePct = Math.min(1 - timeTipPct, Math.max(timeTipPct, pct));
+
+            this.timeTip.el.style.left = (safePct * 100) + '%';
```

Comment:
This should be applied with `utils.css.style` to take advantage of it's check that prevents redundant writes.

Also `toFixed(1)` the percent to limit how frequently the value can change. x1000 precision should be good no?